### PR TITLE
Fix mood runtimes with error handling

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
The runtime log:

```
[2023-09-26 04:34:40.230] RUNTIME: runtime error: Cannot read null._status_traits
 - proc name: set sanity (/datum/mood/proc/set_sanity)
 -   source file: code/datums/mood.dm,417
 -   usr: null
 -   src: /datum/mood (/datum/mood)
 -   call stack:
 - /datum/mood (/datum/mood): set sanity(48.5, 50, 125, 0)
 - /datum/mood (/datum/mood): process(1)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop(2)
 - Master (/datum/controller/master): StartProcessing(0)
 - 
[2023-09-26 04:34:40.231] RUNTIME: runtime error: Cannot read null._status_traits
 - proc name: handle nutrition (/datum/mood/proc/handle_nutrition)
 -   source file: code/datums/mood.dm,116
 -   usr: null
 -   src: /datum/mood (/datum/mood)
 -   call stack:
 - /datum/mood (/datum/mood): handle nutrition()
 - /datum/mood (/datum/mood): process(1)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop(2)
 - Master (/datum/controller/master): StartProcessing(0)
 - 
[2023-09-26 04:34:40.232] RUNTIME: runtime error: Cannot read null._status_traits
 - proc name: process (/datum/mood/process)
 -   source file: code/datums/mood.dm,99
 -   usr: null
 -   src: /datum/mood (/datum/mood)
 -   call stack:
 - /datum/mood (/datum/mood): process(1)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): fire(0)
 - Mood (/datum/controller/subsystem/processing/mood): ignite(0)
 - Master (/datum/controller/master): RunQueue()
 - Master (/datum/controller/master): Loop(2)
 - Master (/datum/controller/master): StartProcessing(0)
 ```

From looking at the [runtime logs for the round](https://tgstation13.download/parsed-logs/manuel/data/logs/2023/09/26/round-215371/runtime.txt), it looks like someone spawned a replica pod mob in null space. (via harvesting seeds with a plant bag)  Then I guess this mob somehow had mood events triggered later on from some method without   Not really sure what happened as the only proc that handles making the `mob_parent` null is when the signal for `COMSIG_QDELETING` is called.  

Regardless, I added an error handling check in `process()` to make sure `mob_parent` exists and if it doesn't to `qdel` and throw a stack trace.

## Why It's Good For The Game
No more runtime mood spam.

## Changelog
:cl:
fix: Fix mood events to not process without a mob assigned to it
/:cl:

### CODE REVIEW

File: code/datums/mood.dm [hunk 1]
```diff
@@ -73,6 +73,11 @@
 	return ..()
 
 /datum/mood/process(seconds_per_tick)
+	if(!mob_parent)
```
Comment (ninjanomnom) [inline code/datums/mood.dm:76]: This is just papering over the issue of either, map_parent ceasing to exist without cleaning up the mood datum, or a mood datum being created without a mob_parent. The core issue should be fixed.

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/datums/mood.dm (hunk 1 above)
Comment (timothymtorres) [reply code/datums/mood.dm:76]: I am unable to replicate what is causing the core issue.  However this solution prevents a lot of runtime errors from spamming the logs. Instead it creates one stack trace which is helpful.

Comment (ninjanomnom) [reply code/datums/mood.dm:76]: Get this testmerged if you'd like to use this to try to find the issue, but this is not a proper fix. Reducing runtimes in the log should not be the end goal.

**Comment by timothymtorres** [comment]
In https://github.com/tgstation/tgstation/pull/78471 the runtime with replica pod mobs spawning in null space was fixed.  That was likely causing the issue so going to close this for now.

**Pull request closed without merging**